include/maidsafe/crux/detail/multiplexer.hpp

```diff
@@ -137,7 +137,8 @@ class multiplexer : public std::enable_shared_from_this<multiplexer>
     using socket_map = std::map<endpoint_type, socket_base *>;
     socket_map sockets;
 
-    std::atomic<int> receive_calls;
+    using receive_calls_underlying_type = int;
+    std::atomic<receive_calls_underlying_type> receive_calls;
```

Comment:
Per email discussion, please change the type to use
```
std::atomic<std::size_t> receive_calls;
```
and remove the `static_cast`s from the assertions below.